### PR TITLE
Quick sysupdate fixups

### PR DESCRIPTION
**Pull request by AdrianVovk**
See commits

### CODE REVIEW

File: src/sysupdate/sysupdated.c [hunk 1]
```diff
@@ -1131,11 +1131,15 @@ static int target_method_vacuum_finish(
...
 
         assert(json);
 
-        instances = sd_json_variant_unsigned(sd_json_variant_by_key(json, "removed"));
+        v = sd_json_variant_by_key(json, "removed");
+        if (!v)
+                return log_sysupdate_bad_json(SYNTHETIC_ERRNO(EPROTO), "vacuum", "Missing key 'removed'");
+        instances = sd_json_variant_unsigned(v);
 
         return sd_bus_reply_method_return(msg, "u", instances);
```
Comment (yuwata) [inline src/sysupdate/sysupdated.c]: When you touch the code, please check the range of `instance` and cast to `uint32_t`, as `"u"` expects `uint32_t`.

**Review by yuwata — APPROVED** [review]
LGTM. One request.

File: src/sysupdate/sysupdated.c [hunk 2]
```diff
@@ -816,16 +816,20 @@ static int sysupdate_run_simple(sd_json_variant **ret, Target *t, ...) {
 
         r = sd_json_parse_file(f, "stdout", 0, &v, NULL, NULL);
         if (r < 0)
-                return log_error_errno(r, "Failed to parse JSON: %m");
+                return log_debug_errno(r, "Failed to parse JSON: %m");
```
Comment (poettering) [inline src/sysupdate/sysupdated.c:819]: hmm, why this change? this should be logged about, no? and given the flags passed here aren't set to SD_JSON_LOG there would be no logging here at all now?

(actually logging here is all over the place?

Comment (AdrianVovk) [reply src/sysupdate/sysupdated.c:819]: Otherwise we log twice about the same error. Once here, and once at some of the call sites of `sysupdate_run_simple`. The commit makes sure that all callsites of `sysupdate_run_simple` log now, and sliences `sysupdate_run_simple`'s own logging to avoid double-logging the errors.

Comment (AdrianVovk) [reply src/sysupdate/sysupdated.c:819]: re: flags, we're not dispatching here, we're parsing. SD_JSON_LOG is a dispatch flag
